Mods/ArchipelagoCore.SC2Mod/Base.SC2Data/Lib5BD4895D.galaxy

```diff
@@ -1149,17 +1287,14 @@ int lib5BD4895D_gf_AP_Core_objectiveIndex (int lp_objective, int lp_categoryInde
     // Variable Initialization
 
     // Implementation
-    if ((lp_categoryIndex >= 0) && (lib5BD4895D_gv_aP_Core_objectivePanel.lv_categories[lp_categoryIndex].lv_rollingIndex > 0)) {

```

Comment:
Not sure we should fully remove this check that `lp_categoryIndex` must be >= 0, because if `AP_Core_objectiveCategoryIndex` fails to find the objective it will return `-1`, and that will generally be passed straight to this function where `lp_categoryIndex` gets used to index into `lv_categories`, which I think will cause a crash. Better to catch that explicitly and early-return `-1` here or something.

Reply:
That's a good point. Fixed. I have also went through and added a few more "if objective index == -1 skip remaining actions" to wherever this was missing to mitigate crashes there too.

Reply:
I've added such check because it broke some maps due to `categoryIndex` being `-1`

That happened if the map has multiple stages and the objectives from previous stages get hidden